.github/workflows/tests.yml

```diff
@@ -0,0 +1,108 @@
+name: Test Suite
+on: [push, pull_request]
```

Comment:
I suggest listening on these events:

```
on:
  push:
    branches:
      - master
  pull_request:
```

This way you don't kick off two sets of tests for a pull_request.